aligned widget with center of screen

epjane opened the pull request:
Adjusted the width of the widget until it was exactly the width needed to draw everything, and therefore it lines itself up with the center of the screen. Needed a different width when showing the seconds. Also made the "off" pixels darker to make it easier to read.

epjane committed:
aligned with center of screen, darker when off

Isopolito commented:
Hi @johanvandegriff , thanks for contributing!

Overall, I like the suggested changes and the implementation is close to correct. Just need to fine tune the width calculations. 

Here's what it looks like on fedora 40.

![Screenshot from 2024-04-28 08-05-26](https://github.com/Isopolito/gBinaryClock/assets/13524742/cad7341f-33dc-4089-b197-4906dd53960c)

Here's what it looks like with seconds displayed:
![Screenshot from 2024-04-28 08-09-56](https://github.com/Isopolito/gBinaryClock/assets/13524742/c38e1de0-ce5d-4641-8064-aea57a275947)

When you get that worked out I'll retest. 

Thanks!

epjane committed:
better align, remove_child on each item

epjane commented:
@Isopolito Thanks for your feedback! I changed it so it is calculated dynamically based on the given width and height. I also added some code to `disable()` that will fully clean it up so the widget doesn't keep moving to the right when enabled/disabled (was even happening on screen lock/unlock too).

Side note: figured out how to use looking glass (Alt+F2 `lg`) which was helpful for inspecting the invisible items left behind, and a nested gnome session so I didn't have to keep logging out: `dbus-run-session -- gnome-shell --nested --wayland`

Isopolito commented:
Looks good, I'm going to merge to main and continue testing. If no issues come up I'll submit a new extension release.

Thanks again for contributing!
